### PR TITLE
Add java12 template

### LINKED ISSUE #146: Java 11 & 12 template
Can't create function with Java 11 or Java 12
## Expected Behaviour

a Java 11 or Java 12 template provided
## Current Behaviour

only Java 8 template available
## Steps to Reproduce (for bugs)

1.
2.
3.
4.

## Your Environment

* Docker version `docker version` (e.g. Docker 17.0.05 ):

* Are you using Docker Swarm or Kubernetes (FaaS-netes)?

* Operating System and version (e.g. Linux, Windows, MacOS):

* Link to your project or a code example to reproduce issue:

### PR DESCRIPTION
**Pull request by alexellis**
Signed-off-by: Alex Ellis <alexellis2@gmail.com>

## Description

The OpenJDK now ships on Oracle Linux 7 with a different EULA
and terms. These have been mentioned in ther welcome message
for unsuspecting users.

Dockerfile needed to change to accomodate the different command
line commands on Oracle Linux vs Alpine Linux. The size has
also increased due to no "light-weight" runtime image being
found or provided at the time.

VM options had to change to -XX:+UseContainerSupport from the
old flags.

Tested on Docker Swarm.

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## Which issue(s) this PR fixes 

Fixes #146 

## How Has This Been Tested?

Tested with simple hello world

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Version change (see: Impact to existing users)

## Impact to existing users

New template, new image, with new Linux base image and EULA. This is printed to users when they run `faas-cli new`

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`

### CODE REVIEW

**Comment by alexellis** [comment]
Thanks for approving @affix 

Next step is to get the libraries used in the Java templates published to a central place, so we can pull them in as a JAR.

File: template/java12/bin/settings.gradle [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/*
+ * This file was generated by the Gradle 'init' task.
+ *
+ * The settings file is used to specify which projects to include in your build.
+ * 
+ * Detailed information about configuring a multi-project build in Gradle can be found
+ * in the user guide at https://docs.gradle.org/4.8.1/userguide/multi_project_builds.html
+ */
+
+rootProject.name = 'java8'
```
Comment (kameshchauhan) [inline template/java12/bin/settings.gradle]: Shouldn't it be changed to java12?

File: template/java12/bin/template.yml [hunk 2]
```diff
@@ -0,0 +1 @@
+language: java8
```
Comment (kameshchauhan) [inline template/java12/bin/template.yml]: Shouldn't it be changed to java12?

**Comment by kameshchauhan** [comment]
@alexellis I would avoid commenting about use of java8 in the few files separately. But in general, there is a need to rectify the use from java8 to java12...mostly in the /bin/*.

File: template/java12/bin/template.yml (hunk 2 above)
Comment (alexellis) [reply template/java12/bin/template.yml]: The bin folder should be removed. This file isnt used

**Comment by alexellis** [comment]
The bin folder should be removed.. my mistake

**Pull request merged**